nix/overlays/torch.nix

```diff
@@ -1,20 +1,49 @@
 final: prev:
 let
-  inherit (prev) lib;
+  inherit (prev) lib system;
   libtorch =
     { device ? "cpu"
     , cudaSupport ? lib.hasPrefix "cuda" device
     }:
-      assert lib.assertOneOf "device" device
-        [
-          "cpu"
-          "cuda-10"
-          "cuda-11"
-        ];
+    # We only care about supporting this platform for now. Darwin support
+    # isn't necessary
+      assert system == "x86_64-linux";
+      assert lib.assertOneOf "device" device [ "cpu" "cuda-118" ];
       assert device == "cpu" -> !cudaSupport;
-      final.callPackage "${final.hasktorch}/nix/libtorch.nix" {
-        inherit device cudaSupport;
-      };
+
+      (prev.libtorch-bin.override { inherit cudaSupport; }).overrideAttrs
```

Comment:
Hasktorch is on a slightly older version of `libtorch-bin` and Pytorch (2.3.0), hence the overrides